Mastodon/Protocol/Provider/DataSourceProvider+StatusTableViewCellDelegate.swift

```diff
@@ -333,46 +288,31 @@ extension StatusTableViewCellDelegate where Self: DataSourceProvider & AuthConte
         guard let pollTableViewDiffableDataSource = statusView.pollTableViewDiffableDataSource else { return }
         guard let firstPollItem = pollTableViewDiffableDataSource.snapshot().itemIdentifiers.first else { return }
         guard case let .option(firstPollOption) = firstPollItem else { return }
-        
-        let managedObjectContext = context.managedObjectContext
-        
-        Task {
-            var _poll: ManagedObjectRecord<Poll>?
-            var _choices: [Int]?
-            
-            try await managedObjectContext.performChanges {
-                guard let poll = firstPollOption.object(in: managedObjectContext)?.poll else { return }
-                _poll = .init(objectID: poll.objectID)
-                
-                guard poll.multiple else { return }
-                
-                // mark voting
-                poll.update(isVoting: true)
-                // set choice
-                _choices = poll.options
-                    .filter { $0.isSelected }
-                    .map { Int($0.index) }
-                
-                poll.update(updatedAt: Date())
-            }
-            
-            // Trigger vote API request for
-            guard let poll = _poll,
-                  let choices = _choices
-            else { return }
-            
+
+        statusView.viewModel.isVoting = true
+
+        Task { @MainActor in
+            let poll = firstPollOption.poll
+
+            let choices = poll.options
+                .filter { $0.isSelected == true }
+                .compactMap { poll.options.firstIndex(of: $0) }
+
             do {
-                _ = try await context.apiService.vote(
-                    poll: poll,
+                let newPoll = try await context.apiService.vote(
```

Comment:
I just tried to vote on a [poll](https://chaos.social/@daniel_bohrer/112274378167176275) and when I pressed "Vote", the spinner appeared but never disappeared. Also I was able to change my vote during that time. This feels weird. I also couldn't see my vote unlike in other clients. Maybe an `update`-somethingsomething is missing somewhere?

Reply:
1. I'll take a look at blocking the UI while a vote is in progress 👍 

2. Hm that's odd, I haven't seen this bug lately. Do you, by any chance, remember on which screen you used to vote?

Reply:
#1 is fixed now. #2 I could not reproduce :-/

Reply:
Re 2: No, I don't, sorry. Most probably it was the Home-timeline.

I'll keep my eyes open, if I encouter it again, I'll open an issue.